staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -15,9 +15,36 @@
 package model
 
 import (
+	"time"
+
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
 	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
 )
 
+const (
+	// the largest request that will be accepted is 3MB
+	// TODO(DangerOnTheRanger): find out where this is originally declared so we don't have two constants
+	maxRequestSizeBytes = 3000000
+	// OpenAPI duration strings follow RFC 3339, section 5.6 - see the comment on maxDatetimeSizeJSON
+	maxDurationSizeJSON = 32
+	// OpenAPI datetime strings follow RFC 3339, section 5.6, and the longest possible
+	// such string is 9999-12-31T23:59:59.999999999Z, which has length 30 - we add 2
+	// to allow for quotation marks
+	maxDatetimeSizeJSON = 32
+	// RFC 3339 durations need at least 1 character for the duration, one character for the unit, and
+	// "P" at the start of the string, e.g, P1W, so with 3 for quotation marks and a comma, that makes 6
+	minDurationSizeJSON = 6
```

Comment:
I don't know if this disagrees with the RFC, but "0" is an allowed duration.

https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/formats.go#L49

https://go.dev/play/p/63rBGEHuqmF

Reply:
Gotcha, thanks for the playground link! I'll adjust the min size accordingly.